### ISSUE TITLE
Functional VHDL model vs Structural VHDL model?

### ISSUE THREAD

**Issue by rlindsberg** [rlindsberg/1331IL-VHDL-Design#3]
No so sure about what lab instruction means...

**Comment by gitgnmn** [comment]
If they mean something with it I can imagine it the difference between describing the components and signals with and/or/nand/nor/xor/not or using more programming like statements in process blocks like I've described the mux in the Adder8 code in alabb1.

**Comment by gitgnmn** [comment]
The process block emulating a mux:
```
pro_case : process (int_c)
    variable INT_C : std_logic;
    begin
       INT_C := int_c;
       case INT_C is
	when '0' => Sum(7 downto 4) <= int_sum_c0(3 downto 0) after 5 ns;
	when '1' => Sum(7 downto 4) <= int_sum_c1(3 downto 0) after 5 ns;
    end case;
end process;
```
[source](https://github.com/rlindsberg/1331IL-VHDL-Design/blob/alabb1/alabb1/Adder8/Adder8.vhd)

**Comment by gitgnmn** [comment]
But then again. The example code from the exercise is showing a full_Adder and a ripple_adder where the code style is the same but the labb1 uses "functional" to describe one and "structural" for the other.

So to summarise. I don't have the faintest. Maybe we should ask in the wiki...?

**Comment by rlindsberg** [comment]
I did some research after I opened the issue. I found a great answer at Quora: [What is the difference between behavioral and structural code in VHDL?](https://www.quora.com/What-is-the-difference-between-behavioral-and-structural-code-in-VHDL) and some explanations at Canvas: [VHDL Ex1](https://kth.instructure.com/courses/6445/files/1173896/download?wrap=1)

**Comment by gitgnmn** [comment]
So in essence. I'll have to redo my mux part of my Adder8.

**Comment by rlindsberg** [comment]
https://github.com/rlindsberg/1331IL-VHDL-Design/blob/b6cd1dbfaa629fc90ea53707bf2843ae7fa63211/alabb1/Adder8/Adder8.vhd#L46-L54

I understand a part of what you mean.. But is this a simulating process? Assignment 1.1, 1.2 are about VHDL models, which I think they mean components. Or am I wrong?

**Comment by gitgnmn** [comment]
All 1.1, 1.2, 1.3 and 1.4 have "Create a structural/functional VHDL-modell for..." in them. I interpret that as that we should also make the code. If the coding style for structural of functional differs is where I'm not sure.

**Comment by rlindsberg** [comment]
We can absolutely ask on Canvas if you still feel confusing!

**Comment by gitgnmn** [comment]
I'd like an example of the two describing the same circuit side by side. It is the syntax rather than the concept that I'm not sure about. 

If I've understood it correct what I've written, and the example code, is structural. But what makes the code functional? Is my case block allowed in structural or is it only allowed in functional? Is my Adder8 a hybrid? (I'll try to make away with the process block anyway.)

**Comment by rlindsberg** [comment]
It's quite difficult to write pure structural model I think..

**Comment by rlindsberg** [comment]
Functional model:
![lec17-10](https://user-images.githubusercontent.com/2918707/45030346-ce00e100-b04b-11e8-991a-ce018b4f0b2a.png)
![lec17-11](https://user-images.githubusercontent.com/2918707/45030353-d0fbd180-b04b-11e8-89fe-b49d108e557c.png)

Structural model:
![lec17-12](https://user-images.githubusercontent.com/2918707/45030388-ecff7300-b04b-11e8-9693-f87400d35249.png)
![lec17-13](https://user-images.githubusercontent.com/2918707/45030396-f12b9080-b04b-11e8-86a5-f8e5cb7f3599.png)
![lec17-14](https://user-images.githubusercontent.com/2918707/45030399-f4268100-b04b-11e8-8161-6955f6a42d34.png)

**Comment by gitgnmn** [comment]
So the only real difference is level of detail. Not really what code you use. Behaviour just described the whole component while structural described all the components in the component. Interesting.

**Comment by rlindsberg** [comment]
Agreed. But the slides are from Northwestern University. Maybe our teacher won't buy it..
